tests/BenchmarkDotNet.IntegrationTests/MemoryDiagnoserTests.cs

```diff
@@ -63,47 +84,89 @@ public MemoryDiagnoserTests(ITestOutputHelper outputHelper)
+            var newArrayBenchmarks = benchmarks.Where(benchmark => benchmark.DisplayInfo.Contains("New"));
+
             const int gen0Index = 0;
 
-            foreach (var listStructEnumeratorBenchmark in listStructEnumeratorBenchmarks)
+            foreach (var benchmark in stackallocBenchmarks)
             {
-                var structEnumeratorGen0Collections = gcCollectionColumns[gen0Index].GetValue(
-                    summary,
-                    listStructEnumeratorBenchmark);
+                var gen0Collections = gcCollectionColumns[gen0Index].GetValue(summary, benchmark);
 
-                Assert.Equal("-", structEnumeratorGen0Collections);
+                Assert.Equal("-", gen0Collections);
             }
 
-            foreach (var listObjectEnumeratorBenchmark in listObjectEnumeratorBenchmarks)
+            foreach (var benchmark in newArrayBenchmarks)
+            {
+                var gen0Str = gcCollectionColumns[gen0Index].GetValue(summary, benchmark);
+
+                AssertParsed(gen0Str, gen0Value => gen0Value > 0);
+            }
+        }
+
+        [Fact(Skip = "Temporarily suppressed, see https://github.com/PerfDotNet/BenchmarkDotNet/issues/208")]
+        public void MemoryDiagnoserDoesNotIncludeAllocationsFromSetupAndCleanup()
+        {
+            AssertZeroAllocations(typeof(AllocatingSetupAndCleanup), "AllocateNothing", targetCount: 50);
+        }
+
+        [Fact(Skip = "Temporarily suppressed, see https://github.com/PerfDotNet/BenchmarkDotNet/issues/208")]
+        public void EngineShouldNotInterfereAllocationResults()
+        {
+            AssertZeroAllocations(typeof(NoAllocationsAtAll), "EmptyMethod", targetCount: 5000); // we need a lot of iterations to be sure!!
+        }
+
+        public void AssertZeroAllocations(Type benchmarkType, string benchmarkMethodName, int targetCount)
+        {
+            var memoryDiagnoser = new MemoryDiagnoser();
+            var config = CreateConfig(memoryDiagnoser, targetCount);
+            var benchmarks = BenchmarkConverter.TypeToBenchmarks(benchmarkType, config);
+
+            var summary = BenchmarkRunner.Run((Benchmark[])benchmarks, config);
+
+            var allocationColumn = GetColumns<MemoryDiagnoser.AllocationColumn>(memoryDiagnoser).Single();
+            var allocateNothingBenchmarks = benchmarks.Where(benchmark => benchmark.DisplayInfo.Contains(benchmarkMethodName));
+
+            foreach (var benchmark in allocateNothingBenchmarks)
+            {
+                var allocations = allocationColumn.GetValue(summary, benchmark);
+
+                AssertParsed(allocations, allocatedBytes => allocatedBytes == 0);
+            }
+        }
+
+        private IConfig CreateConfig(IDiagnoser diagnoser, int targetCount)
+        {
+            return ManualConfig.CreateEmpty()
+                               .With(Job.Dry.WithLaunchCount(1).WithWarmupCount(1).WithTargetCount(targetCount).With(GcMode.Default.WithForce(false)))
```

Comment:
This line is too long. Could we introduce a variable for job here?
Also, you could use `WithGcForce`.